diff --git a/drivers/soc/mediatek/mtk-mutex.c b/drivers/soc/mediatek/mtk-mutex.c
index 621ebed0b7..ef702f5861 100644
--- a/drivers/soc/mediatek/mtk-mutex.c
+++ b/drivers/soc/mediatek/mtk-mutex.c
@@ -397,7 +397,7 @@ static const u8 mt2712_mutex_mod[DDP_COMPONENT_ID_MAX] = {
 	[DDP_COMPONENT_WDMA1] = MT2712_MUTEX_MOD_DISP_WDMA1,
 };
 
-static const unsigned int mt6765_mutex_mod[DDP_COMPONENT_ID_MAX] = {
+static const u8 mt6765_mutex_mod[DDP_COMPONENT_ID_MAX] = {
 	[DDP_COMPONENT_AAL0] = 13,
 	[DDP_COMPONENT_CCORR] = 12,
 	[DDP_COMPONENT_COLOR0] = 11,
@@ -411,7 +411,6 @@ static const unsigned int mt6765_mutex_mod[DDP_COMPONENT_ID_MAX] = {
 };
 
 static const u8 mt8167_mutex_mod[DDP_COMPONENT_ID_MAX] = {
->>>>>>> 3d4266f12508 (soc: mediatek: mutex: Add support for MT6765)
 	[DDP_COMPONENT_AAL0] = MT8167_MUTEX_MOD_DISP_AAL,
 	[DDP_COMPONENT_CCORR] = MT8167_MUTEX_MOD_DISP_CCORR,
 	[DDP_COMPONENT_COLOR0] = MT8167_MUTEX_MOD_DISP_COLOR,
